flink-ml-lib/src/test/java/org/apache/flink/ml/feature/SQLTransformerTest.java

```diff
@@ -0,0 +1,241 @@
+        DataStream<Row> inputStream =
+                env.fromCollection(
+                        INPUT_DATA, new RowTypeInfo(Types.INT, Types.DOUBLE, Types.DOUBLE));
+        inputTable = tEnv.fromDataStream(inputStream).as("id", "v1", "v2");
+    }
+
+    @Test
+    public void testParam() {
+        SQLTransformer sqlTransformer = new SQLTransformer();
+        sqlTransformer.setStatement("SELECT * FROM __THIS__");
+        assertEquals("SELECT * FROM __THIS__", sqlTransformer.getStatement());
+    }
+
+    @Test
+    public void testInvalidSQLStatement() {
+        SQLTransformer sqlTransformer = new SQLTransformer();
+
+        try {
+            sqlTransformer.setStatement("SELECT * FROM __THAT__");
+            fail();
+        } catch (Exception e) {
+            assertEquals(
+                    "Parameter statement is given an invalid value SELECT * FROM __THAT__",
+                    e.getMessage());
+        }
+    }
+
+    @Test
+    public void testOutputSchema() {
+        SQLTransformer sqlTransformer =
+                new SQLTransformer()
+                        .setStatement("SELECT *, (v1 + v2) AS v3, (v1 * v2) AS v4 FROM __THIS__");
+
+        Table outputTable = sqlTransformer.transform(inputTable)[0];
+
+        assertEquals(
+                Arrays.asList("id", "v1", "v2", "v3", "v4"),
+                outputTable.getResolvedSchema().getColumnNames());
+    }
+
+    @Test
+    public void testTransformNumericData() {
+        SQLTransformer sqlTransformer =
+                new SQLTransformer()
+                        .setStatement("SELECT *, (v1 + v2) AS v3, (v1 * v2) AS v4 FROM __THIS__");
+
+        Table outputTable = sqlTransformer.transform(inputTable)[0];
+
+        verifyOutputResult(outputTable, EXPECTED_NUMERIC_DATA_OUTPUT);
+    }
+
+    @Test
+    public void testBuiltInFunction() {
+        SQLTransformer sqlTransformer =
+                new SQLTransformer().setStatement("SELECT *, SQRT(v1) AS v3 FROM __THIS__");
+
+        Table outputTable = sqlTransformer.transform(inputTable)[0];
+
+        verifyOutputResult(outputTable, EXPECTED_BUILT_IN_FUNCTION_OUTPUT);
+    }
+
+    @Test
+    public void testBatchAggregation() {
+        env.setRuntimeMode(RuntimeExecutionMode.BATCH);
+        tEnv = StreamTableEnvironment.create(env);
+        DataStream<Row> inputStream =
+                env.fromCollection(
+                        INPUT_DATA, new RowTypeInfo(Types.INT, Types.DOUBLE, Types.DOUBLE));
+        inputTable = tEnv.fromDataStream(inputStream).as("id", "v1", "v2");
+
+        SQLTransformer sqlTransformer =
+                new SQLTransformer().setStatement("SELECT SUM(v1) AS v3 FROM __THIS__ GROUP BY v2");
+
+        Table outputTable = sqlTransformer.transform(inputTable)[0];
+
+        verifyOutputResult(outputTable, EXPECTED_BATCH_AGGREGATION_OUTPUT, false);
+    }
+
+    @Test
+    public void testStreamingAggregation() {
+        env.setRuntimeMode(RuntimeExecutionMode.STREAMING);
```

Comment:
Runtime mode is streaming by default. It seems better to remove this line for consistency with existing tests.